flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,593 @@
+        self.log = logger
+        self.kwargs = kwargs
+
+    def __str__(self):
+        return self.value
+
+
+class RadarrMovieAlreadyExistsError(Exception):
+    def __init__(self):
+        super(RadarrMovieAlreadyExistsError, self).__init__()
+
+
+def spec_exception_from_response_ex(radarr_request_ex):
+    error_message = None
+
+    if "error_message" in radarr_request_ex.kwargs:
+        error_message = radarr_request_ex.kwargs["error_message"]
+
+    if not error_message:
+        return None
+
+    if error_message.lower() == "this movie has already been added":
+        return RadarrMovieAlreadyExistsError()
+
+    return None
+
+
+def request_get_json(url, headers):
+    """ Makes a GET request and returns the JSON response """
+    try:
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise RadarrRequestError(
+                "Invalid response received from Radarr: %s" % response.content
+            )
+    except RequestException as e:
+        raise RadarrRequestError("Unable to connect to Radarr at %s. Error: %s" % (url, e))
+
+
+def request_delete_json(url, headers):
+    """ Makes a DELETE request and returns the JSON response """
+    try:
+        response = requests.delete(url, headers=headers)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise RadarrRequestError(
+                "Invalid response received from Radarr: %s" % response.content
+            )
+    except RequestException as e:
+        raise RadarrRequestError("Unable to connect to Radarr at %s. Error: %s" % (url, e))
+
+
+def request_post_json(url, headers, data):
+    """ Makes a POST request and returns the JSON response """
+    try:
+        response = requests.post(url, headers=headers, data=data)
+        if response.status_code == 201:
+            return response.json()
+        else:
+            error_message = None
+            try:
+                json_response = response.json()
+                if len(json_response) > 0 and "errorMessage" in json_response[0]:
+                    error_message = json_response[0]["errorMessage"]
+            except ValueError:
+                # Raised by response.json() if JSON couln't be decoded
+                log.error("Radarr returned non-JSON error result: %s", response.content)
+
+            raise RadarrRequestError(
+                "Invalid response received from Radarr: %s" % response.content,
+                log,
+                status_code=response.status_code,
+                error_message=error_message,
+            )
+
+    except RequestException as e:
+        raise RadarrRequestError("Unable to connect to Radarr at %s. Error: %s" % (url, e))
+
+
+def request_put_json(url, headers):
+    """ Makes a PUT request and returns the JSON response """
+    try:
+        response = requests.put(url, headers=headers)
```

Comment:
See above.

It also seems all these requests methods basically do the same, so would it make sense to create a single function that takes in a request type as argument?

Reply:
Agreed, I did not really like these.